src/planscape/datasets/models.py

```diff
@@ -0,0 +1,96 @@
+from django.contrib.postgres.fields import ArrayField
+from django.contrib.auth import get_user_model
+from django.contrib.gis.db import models
+from django.conf import settings
+from core.models import CreatedAtMixin, UpdatedAtMixin, DeletedAtMixin, UUIDMixin
+
+
+User = get_user_model()
+
+
+class DatasetType(models.TextChoices):
+
+    VECTOR = "VECTOR", "Vector"
+    RASTER = "RASTER", "Raster"
+
+
+class DatasetBlobStatus(models.TextChoices):
+    """
+    Represents an enumeration of possible dataset blob statuses.
+    This is about the life-cycle of the dataset.
+    """
+
+    PENDING_UPLOAD = "PENDING_UPLOAD", "Pending Upload"
+    READY = "READY", "Ready"
+
+
+class Dataset(
+    UUIDMixin,
+    CreatedAtMixin,
+    UpdatedAtMixin,
+    DeletedAtMixin,
+    models.Model,
+):
+    """
+    Represents a GIS dataset that can be used in a metric
+    or in a restriction area.
+    """
+
+    owner = models.ForeignKey(
+        User,
+        on_delete=models.RESTRICT,
+        related_name="owned_datasets",
+    )
+
+    organization = models.ForeignKey(
+        Organization,
+        on_delete=models.RESTRICT,
+        related_name="datasets",
+    )
+
+    name = models.CharField(max_length=256)
+
+    type = models.CharField(
+        choices=DatasetType.choices,
+    )
+
+    blob_status = models.CharField(
+        choices=DatasetBlobStatus.choices,
+        default=DatasetBlobStatus.PENDING_UPLOAD,
+    )
+
+    info = models.JSONField(
+        null=True,
+        help_text="Contains the result of ogrinfo or gdalinfo in this dataset. This has important information on how the dataset behaves. Can only be obtained after the dataset is in storage.",
```

Comment:
Can we annotate this json object a bit further? 

> This has important information on how the dataset behaves

If we know some properties already it would be great to have them documented somewhere